### DISCUSSION TITLE
adding two fields in METplus config

### DISCUSSION THREAD

**Discussion by Deba57** [dtcenter/METplus#2081] [Statistical Computation] [answered]
Hi,
I have two model forecast fields, convective and stratiform precipitation in grib2 file and would like to add them to get total precipitation to validate against observation (which is total precipitation). Using Gridstat config file in MetPlus, how can I add these two model fields to compute total precipitation ? Or do I need to compute that offline ? 
Thanks
Debasish .

**Comment by CPKalb** [comment, marked as answer]
Hi Debasish:

Adding two fields can be performed using MET's Pcp-Combine tool.  It can't be done directly in Grid-Stat.  Since you are adding fields with different variable names, you will need to use the User Defined option for Pcp-Combine in METplus.  The setup would include the information below.

FCST_PCP_COMBING_RUN = True
FCST_PCP_COMBINE_METHOD = USER_DEFINED

FCST_PCP_COMBINE_COMMAND = -add {FCST_PCP_COMBINE_INPUT_DIR}/{FCST_PCP_COMBINE_INPUT_TEMPLATE} 'name="convective_var_name"; level="convective_var_level";' {FCST_PCP_COMBINE_INPUT_DIR}/{FCST_PCP_COMBINE_INPUT_TEMPLATE} 'name="stratiform_var_name"; level="stratiform_var_level";'
 
You will also need to set the FCST_PCP_COMBINE_INPUT_DIR, FCST_PCP_COMBINE_INPUT_TEMPLATE, FCST_PCP_COMBINE_OUTPUT_DIR, and FCST_PCP_COMBINE_OUTPUT_TEMPLATE. You should be able to put the Grid-Stat and Pcp-Combine options in the same METplus configuration file.  I'm including some links below that show examples of setting up the User Defined use case, as well as information about Pcp-Combine.

[PCP-Combine User Defined basic use case](https://metplus.readthedocs.io/en/latest/generated/met_tool_wrapper/PCPCombine/PCPCombine_user_defined.html)
[Example showing both PCP-Combine and Grid-Stat in a configuration file](https://metplus.readthedocs.io/en/latest/generated/model_applications/precipitation/GridStat_fcstGFS_obsCCPA_GRIB.html)
[PCPCombine MET Guide](https://met.readthedocs.io/en/latest/Users_Guide/reformat_grid.html#pcp-combine-tool)

Let me know if you have additional questions.

Christina

**Reply by Deba57** [reply]
Thanks @CPKalb for your reply. I have another question on data format. Model I am using is HRRR and it uses curvilinear grid and observation is regular lat lon (over globe). Do I have to set regrid differently between these two grids in METplus ? Following is lat/lon grid info for the model : 

float gridrot_0 ( ygrid_0, xgrid_0 )
         long_name :	vector rotation angle
         GridType :	Lambert Conformal (secant, tangent, conical or bipolar)
         units :	radians
         formula_u :	Ugrid = cos(rot)*Uearth - sin(rot)*Vearth
         formula_v :	Vgrid = sin(rot)*Uearth + cos(rot)*Vearth
         note1 :	u and v components of vector quantities are resolved relative to earth
         note2 :	apply formulas to derive u and v components relative to grid

      float gridlat_0 ( ygrid_0, xgrid_0 )
         corners :	( 21.13812, 21.14055, 47.84219, 47.83862 )
         long_name :	latitude
         grid_type :	Lambert Conformal (secant, tangent, conical or bipolar)
         units :	degrees_north
         Latin2 :	38.5
         Latin1 :	38.5
         Dy :	 3
         Dx :	 3
         Lov :	262.5
         Lo1 :	237.2805
         La1 :	21.13812

      float gridlon_0 ( ygrid_0, xgrid_0 )
         corners :	( -122.7195, -72.28972, -60.91719, -134.0955 )
         long_name :	longitude
         grid_type :	Lambert Conformal (secant, tangent, conical or bipolar)
         units :	degrees_east
         Latin2 :	38.5
         Latin1 :	38.5
         Dy :	 3
         Dx :	 3
         Lov :	262.5
         Lo1 :	237.2805
         La1 :	21.13812

Thanks
Deba

**Reply by CPKalb** [reply]
Hi Deba:

They will need to be on the same grid for verification, but there are options in how this is done.  In all of these cases, MET can perform the regridding either by using the regrid options in Grid-Stat, or by running MET's regrid_data_plane tool.   You could perform the regridding differently for the model and observations, but you don't have to do that.  It really depends on what you want for output, and whether it makes sense to regrid them the same way or to do them differently. 

For example, you could put the observation on the model grid by setting FCST as the GRID_STAT_REGRID_TO_GRID, or you could put the model on the observation grid by selecting OBS as the GRID_STAT_REGRID_TO_GRID.   The third option is to put them both on a different grid, and specify that as the GRID_STAT_REGRID_TO_GRID.  MET has options in how regridding is done, such as bilinear, nearest neighbor, least squares, and many more.  I'm including some links to regridding options and examples of how to set up these different configurations in METplus.

[Example of regridding both the model and observations to a different grid inside Grid-Stat (the G002 grid)](https://metplus.readthedocs.io/en/latest/generated/model_applications/medium_range/GridStat_fcstGFS_obsGFS_Sfc_MultiField.html)
[Example of regridding to the forecast grid inside grid stat](https://metplus.readthedocs.io/en/latest/generated/model_applications/climate/GridStat_fcstCESM_obsGFS_ConusTemp.html)
[Example using regrid data plane](https://metplus.readthedocs.io/en/latest/generated/met_tool_wrapper/RegridDataPlane/RegridDataPlane_multi_field_multi_file.html)
[Regrid data plane information](https://met.readthedocs.io/en/latest/Users_Guide/reformat_grid.html#regrid-data-plane-tool)
[Grid stat regridding information, section 5.2.10](https://met.readthedocs.io/en/latest/Users_Guide/config_options.html#settings-common-to-multiple-tools)

Does this help?

Christina

**Comment by CPKalb** [comment]
Hi Deba:

I wanted to check back with you.  Were you able to get this up and running as expected?

Thanks,
Christina
